### PR TITLE
dot/types: Update authority weights to 1, create new Authority type

### LINKED ISSUE #1052: update authority weights to 1 in genesis, update types.GrandpaAuthorityData and types.BABEAuthorityData
## Expected Behavior

- see spec 6.1.1. Consensus Authority Set
- authorities for both BABE and grandpa should be:
```
type Authority struct {
     Key // sr25519 for BABE, ed25519 for grandpa
     Weight // always set to 1 currently
}
```

we can likely combine the existing `GrandpaAuthorityData` and `BABEAuthorityData` types to use `crypto.PublicKey` for the `key` type

## Checklist

- [x] I have read [CODE_OF_CONDUCT](https://github.com/ChainSafe/gossamer/blob/development/.github/CODE_OF_CONDUCT.md) and [CONTRIBUTING](https://github.com/ChainSafe/gossamer/blob/development/.github/CONTRIBUTING.md) 
- [x] I have provided as much information as possible and necessary
- [ ] I am planning to submit a pull request to fix this issue myself

### PR DESCRIPTION
**Pull request by edwardmack**
## Changes

- update authority weights to 1 in genesis config files.
- Implement type `types.Authority` to replace `types.GrandpaAuthorityData` and `types.BABEAuthorityData`
-

## Tests

```
go test ./... -short
```

## Checklist

- [x] I have read [CODE_OF_CONDUCT](https://github.com/ChainSafe/gossamer/blob/development/.github/CODE_OF_CONDUCT.md) and [CONTRIBUTING](https://github.com/ChainSafe/gossamer/blob/development/.github/CONTRIBUTING.md) 
- [x] I have provided as much information as possible and necessary
- [x] I have reviewed my own pull request before requesting a review
- [x] All integration tests and required coverage checks are passing

## Issues

- closes #1052

### CODE REVIEW

File: dot/types/babe.go [hunk 1]
```diff
@@ -15,19 +13,19 @@ type BabeConfiguration struct {
...
 
 // BABEAuthorityDataRaw represents a BABE authority where their key is a byte array
-type BABEAuthorityDataRaw struct {
-	ID     [sr25519.PublicKeyLength]byte
-	Weight uint64
-}
+//type BABEAuthorityDataRaw struct {
+//	ID     [sr25519.PublicKeyLength]byte
+//	Weight uint64
+//}
```
Comment (noot) [inline dot/types/babe.go:25]: remove?

File: dot/types/babe.go [hunk 2]
```diff
@@ -15,19 +13,19 @@ type BabeConfiguration struct {
...
 
-// Decode will decode the Reader into a BABEAuthorityDataRaw
-func (a *BABEAuthorityDataRaw) Decode(r io.Reader) (*BABEAuthorityDataRaw, error) {
+// Decode will decode the Reader into a AuthorityRaw
+func (a *AuthorityRaw) Decode(r io.Reader) (*AuthorityRaw, error) {
```
Comment (noot) [inline dot/types/babe.go:28]: maybe move this to `authority.go`?

File: lib/grandpa/grandpa.go [hunk 3]
```diff
@@ -170,25 +170,29 @@ func (s *Service) Stop() error {
...
 }
 
 // UpdateAuthorities schedules an update to the grandpa voter set and increments the setID at the end of the current round
-func (s *Service) UpdateAuthorities(ad []*types.GrandpaAuthorityData) {
+func (s *Service) UpdateAuthorities(ad []*types.Authority) {
 	v := make([]*Voter, len(ad))
 	for i, a := range ad {
+		ed25519PK, err := ed25519.NewPublicKey(a.Key.Encode())
+		if err != nil {
+			// todo determine how to handle this error
```
Comment (noot) [inline lib/grandpa/grandpa.go:191]: make this function return an error and then handle it in whatever is calling it?

File: lib/grandpa/grandpa_test.go [hunk 4]
```diff
@@ -114,8 +114,8 @@ func TestUpdateAuthorities(t *testing.T) {
 		{key: kr.Alice().Public().(*ed25519.PublicKey), id: 0},
```
Comment (noot) [inline lib/grandpa/grandpa_test.go:114]: change `id` to `Weight`?

File: lib/grandpa/message_handler.go [hunk 5]
```diff
@@ -310,7 +310,11 @@ func (h *MessageHandler) verifyJustification(just *Justification, vote *Vote, ro
 	// verify authority in justification set
 	authFound := false
 	for _, auth := range h.grandpa.Authorities() {
-		if reflect.DeepEqual(auth.Key.AsBytes(), just.AuthorityID) {
+		justKey, err := just.AuthorityID.Encode()
+		if err != nil {
+			return err
+		}
+		if reflect.DeepEqual(auth.Key.Encode(), justKey) {
```
Comment (noot) [inline lib/grandpa/message_handler.go:317]: why is this changed?

File: lib/grandpa/types.go [hunk 6]
```diff
@@ -79,13 +79,17 @@ func (v *Voter) String() string {
...
 // NewVotersFromAuthorityData returns an array of Voters given an array of GrandpaAuthorityData
-func NewVotersFromAuthorityData(ad []*types.GrandpaAuthorityData) []*Voter {
+func NewVotersFromAuthorityData(ad []*types.Authority) []*Voter {
 	v := make([]*Voter, len(ad))
 
 	for i, d := range ad {
+		ed25519PK, err := ed25519.NewPublicKey(d.Key.Encode())
+		if err != nil {
+			// todo determine how to handle this error
+		}
```
Comment (noot) [inline lib/grandpa/types.go:89]: you should do an interface conversion here:
```
if pk, ok := d.key.(*ed25519.PublicKey); ok {
     v[i] = ...
}
```

**Review by noot — COMMENTED** [review]
overall looking good!! just a few comments

File: lib/grandpa/message_handler.go (hunk 5 above)
Comment (edwardmack) [reply lib/grandpa/message_handler.go:317]: DeepEqual was returning false because the types don't match.  I used to work with `auth.Key.AsBytes()`, but AsBytes no longer exists.  So I changed so justKey type matchs auth.Key.Encode ([]uint8)

**Comment by edwardmack** [comment]
I've addressed the comments, I think it's ready now, let me know.

**Review by noot — APPROVED** [review]
great work!
